src/decoder/stream.rs

```diff
@@ -629,6 +651,42 @@ impl StreamingDecoder {
                             ))
                         }
                     }
+                    ApngSequenceNumber => {
+                        debug_assert_eq!(self.current_chunk.type_, chunk::fdAT);
+                        let next_seq_no = val;
+                        if self.current_chunk.remaining < 4 {
```

Comment:
I think it would be more natural to check this before consuming 4 bytes of input, given that some of those bytes correspond to the length field of the next chunk!

Reply:
Done (leaving a `debug_assert!` here).